stdlib/Printf/src/Printf.jl

```diff
@@ -158,16 +160,16 @@ function Format(f::AbstractString)
             b = bytes[pos]
             pos += 1
             if b == prev
-                pos > len && throw(ArgumentError("invalid format string: '$f'"))
+                pos > len && throw(ArgumentError("Format string is invalid - unterminated length modifier at end of string: '$f'"))
                 b = bytes[pos]
                 pos += 1
             end
-        elseif b in b"Ljqtz"
+        elseif b in b"Ljqtz" # what is q? Possibly quad?
```

Comment:
`q` is not described in the [linked reference doc](http://www.cplusplus.com/reference/cstdio/printf/).

Reply:
I have dug through git history and travelled through the ages, [back when julia files ended in `.j`](https://github.com/JuliaLang/julia/blame/773bf1bf8e1f67706b8857f69e8cbecc84fee245/j/printf.j#L46).. I don't think @StefanKarpinski will remember what `q` did, but I guess it can't hurt to ask? :sweat_smile: 

I do have the suspicion that it's from the `printf` shell command, not the linked C++ doc:

```
 %q     ARGUMENT  is  printed  in  a format that can be reused as shell input, escaping non-printable characters with the proposed POSIX $''
        syntax.
```

If so, I'll add that as a comment, to prevent future confusion.

Reply:
This is parsed as a size prefix, so
``` 
       q      A synonym for ll.  This is a nonstandard extension,
              derived from BSD; avoid its use in new code.
```
seems to fit well.

Reply:
Makes sense! I'll add that as a comment.